### DISCUSSION TITLE
Quarto on juptyer notebooks is not executing iPython startup files?

### DISCUSSION THREAD

**Discussion by astrowonk** [quarto-dev/quarto-cli#7130] [Q&A]
### Description

When Jupyterlab or Jupyter notebook server starts up a .ipynb file, or when ipython is launched, it first runs everything in `~/.ipython/profile_default/startup/`. This is very convenient as a lot of code and imports don't have to be included in every notebook.

I can't quite tell what quarto is doing when I `--execute` a notebook.  Some cells that I _thought_ would fail without what's in the startup files seem to run, but then others fail with missing import errors that seem to mean that the startup ipython files haven't been run.

In any event, an argument to execute the [startup scripts](https://ipython.org/ipython-doc/1/config/overview.html#startup-files) would be helpful, and or to make this default behavior. It would be very inconvenient to use quarto on notebooks where I expect the startup files to have run: I'd have to manually move in a bunch of imports into each notebook.

**Comment by cscheid** [comment]
> I'd have to manually move in a bunch of imports into each notebook.

We intend quarto's notebook features to be useful for reproducibility of computational experiments. I agree with you that it's less convenient, but on the balance, we tend to choose on the side of explicitly declaring such dependencies. For example, we do this not only for jupyter notebooks, but we do it for quarto extensions more generally.

In short: we designed this behavior intentionally.
